app/src/lib/helpers/repo-rules.ts

```diff
@@ -12,6 +12,55 @@ import {
   IAPIRepoRuleMetadataParameters,
   IAPIRepoRuleset,
 } from '../api'
+import { enableRepoRulesBeta } from '../feature-flag'
+import { supportsRepoRules } from '../endpoint-capabilities'
+import { Account } from '../../models/account'
+import { Repository } from '../../models/repository'
+import { GitHubRepository } from '../../models/github-repository'
+
+/**
+ * Returns whether repo rules could potentially exist for the provided account and repository.
+ * This only performs client-side checks, such as whether the user is on a free plan
+ * and the repo is public.
+ */
+export function useRepoRulesLogic(
```

Comment:
I think we can simplify this method a bit by making changing the repository parameter to `repository: Repository` and using the `isRepositoryWithGitHubRepository` help method to identify if the repo has a GitHubRepository object on it.

Here shows what I am talking about:
https://github.com/desktop/desktop/compare/40017cac396ce76558ebb70a8724be7b6e765203...Use-isRepositoryWithGitHubRepository-Helper

Reply:
Good call, I didn't know about that helper, thanks! Just pushed that.

Reply:
2e5f7947e18e178a0494f28f1ccb7bc798a5aafa